protolog/loglistener.go

```diff
@@ -144,86 +132,123 @@ func (ll *LogListener) Shutdown() {
 	close(ll.logEntriesRecieved)
 }
 
+func (ll *LogListener) processConnection(conn net.Conn) {
+	if ll.config.JsonMode {
+		buffer := Buffer{conn, make([]byte, 0)}
+		dec := jstream.NewDecoder(&buffer, 0)
+		for log := range dec.Stream() {
+			ll.processJson(log.Value.(map[string]interface{}), buffer.String())
```

Comment:
I'm torn between maintaining BC and dropping support for schema validation. Removing schema validation support would remove the need for the buffer and reduce memory usage and memory copying, but would prevent us from pushing changes upstream.